wrappers/services.go

```diff
@@ -377,8 +377,15 @@ func ServicesEnableState(s *snap.Info, inter interacter) (map[string]bool, error
 	return snapSvcsState, nil
 }
 
-// RemoveSnapServices disables and removes service units for the applications from the snap which are services.
-func RemoveSnapServices(s *snap.Info, inter interacter) error {
+// RemoveSnapServices disables and removes service units for the applications
+// from the snap which are services. The optional flag indicates whether
+// services are removed as part of undoing of first install of a given snap.
+func RemoveSnapServices(s *snap.Info, firstInstall bool, inter interacter) error {
+	if s.GetType() == snap.TypeSnapd && firstInstall {
```

Comment:
At this point this is really getting not that transparent anymore, I think it's probably time to move the == snap.TypeSnapd checks from wrappers into snapstate and have the related helpers be explicit/exported. This will give us a chance to do something about them with snapstate/policy at some point.